### PR TITLE
[#22] Receive file from client

### PR DESCRIPTION
**Pull request by GuChad369**
After trying several methods to send a file from the client, I will now use our extension to receive the file from the client. These are functions on the extension side, which will be called from the client side to send the file chunks. The logic is very similar to the `pgmoneta_ext_get_file` function.

### CODE REVIEW

**Comment by jesperpedersen** [comment]
Would be better to do this in one function - as there could be concurrent operations using this API

**Comment by GuChad369** [comment]
The logic of using three functions is clearer than relying on just one. I am currently resolving concurrency issues to ensure that the `FileTransferContext` structure is isolated in different sessions. If necessary, we can also implement a locking mechanism.

**Comment by GuChad369** [comment]
I combined the separate functions into a single function, making it simpler to manage.

File: sql/pgmoneta_ext--0.1.0.sql [hunk 1]
```diff
@@ -24,10 +24,14 @@ CREATE FUNCTION pgmoneta_ext_get_oids() RETURNS SETOF RECORD
...
+CREATE FUNCTION pgmoneta_ext_get_file(file_path text) RETURNS text
 AS 'MODULE_PATHNAME'
 LANGUAGE C STRICT;
 
-CREATE OR REPLACE FUNCTION pgmoneta_ext_get_files(file_path TEXT) RETURNS text[]
+CREATE OR REPLACE FUNCTION pgmoneta_ext_get_files(file_path text) RETURNS text[]
+AS 'MODULE_PATHNAME'
+LANGUAGE C STRICT;
+
+CREATE FUNCTION pgmoneta_ext_receive_file_chunk(base64_chunk text, file_path text) RETURNS int
```
Comment (jesperpedersen) [inline sql/pgmoneta_ext--0.1.0.sql:35]: You need to add this function to the documentation

**Review by jesperpedersen — CHANGES_REQUESTED** [review]

**Comment by jesperpedersen** [comment]
Merged.

Thanks for your contribution !

**Pull request merged**